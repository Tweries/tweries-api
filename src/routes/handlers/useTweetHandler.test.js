const fetch = require('node-fetch');
const makeT = require('../makeT');
const useTweetHandler = require('./useTweetHandler');

jest.mock('../makeT');

describe('useTweetHandler', () => {
  it('response w/o error', async () => {
    fetch.mockResponses(
      [
        JSON.stringify({ access_token: 'mock_access_token' }),
        { status: '200' }
      ],
      [
        JSON.stringify({
          identities: [
            {
              access_token: 'mock_access_token',
              access_token_secret: 'mock_access_token_secret',
              provider: 'twitter'
            }
          ]
        }),
        { status: 200 }
      ]
    );

    makeT.mockImplementation(() => ({ get: jest.fn() }));

    const dictionary = { env: 'test' };

    const req = {
      app: { get: key => dictionary[key] },
      body: {
        statusId: '1200471390236766208',
        userId: 'twitter|1183836409850814464'
      }
    };

    const mockSend = jest.fn();
    const res = { send: mockSend };

    await useTweetHandler(req, res);

    expect(mockSend.mock.calls[0][0]).toMatchSnapshot({
      timestamp: expect.any(Number)
    });
  });

  it('response w/ error', async () => {
    fetch.mockResponses(
      [
        JSON.stringify({ access_token: 'mock_access_token' }),
        { status: '200' }
      ],
      [
        JSON.stringify({
          identities: [
            {
              access_token: 'mock_access_token',
              access_token_secret: 'mock_access_token_secret',
              provider: 'twitter'
            }
          ]
        }),
        { status: 200 }
      ]
    );

    makeT.mockImplementation(() => ({
      get: jest.fn(() => {
        throw new Error('Oh Noes!');
      })
    }));

    const dictionary = { env: 'test' };

    const req = {
      app: { get: key => dictionary[key] },
      body: {
        statusId: '1200471390236766208',
        userId: 'twitter|1183836409850814464'
      }
    };

    const mockSend = jest.fn();
    const res = { send: mockSend };

    await useTweetHandler(req, res);

    expect(mockSend.mock.calls[0][0]).toMatchSnapshot({
      timestamp: expect.any(Number)
    });
  });
});
