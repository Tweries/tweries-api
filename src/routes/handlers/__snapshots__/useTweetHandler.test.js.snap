// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTweetHandler response w/ error 1`] = `
Object {
  "env": "test",
  "error": [Error: Oh Noes!],
  "message": "Oh Noes!",
  "name": "tweries-api",
  "statusId": undefined,
  "timestamp": Any<Number>,
  "tweetUrl": "https://twitter.com/musk_china/status/1199474666412236800",
  "userId": "twitter|1183836409850814464",
  "version": "0.11.2",
}
`;

exports[`useTweetHandler response w/o error 1`] = `
Object {
  "env": "test",
  "error": undefined,
  "message": "test",
  "name": "tweries-api",
  "statusId": "1199474666412236800",
  "timestamp": Any<Number>,
  "tweetUrl": "https://twitter.com/musk_china/status/1199474666412236800",
  "userId": "twitter|1183836409850814464",
  "version": "0.11.2",
}
`;
